packages/alfa-style/src/node/predicate/is-transparent.ts

```diff
@@ -23,33 +49,11 @@ export function isTransparent(
     return cache
       .get(device, Cache.empty)
       .get(context, Cache.empty)
-      .get(node, () => {
-        if (
-          isElement(node) &&
-          Style.from(node, device, context)
-            .computed("opacity")
-            .some((opacity) => opacity.value === 0)
-        ) {
-          return true;
-        }
-
-        for (const parent of node.parent(Node.flatTree)) {
-          if (
-            isText(node) &&
-            isElement(parent) &&
-            Style.from(parent, device, context)
-              .computed("color")
-              .some(
-                (color) => color.type === "color" && Color.isTransparent(color)
-              )
-          ) {
-            return true;
-          }
-
-          return isTransparent(parent);
-        }
-
-        return false;
-      });
+      .get(node, () =>
+        or(
+          and(isElement, isNotOpaque(device, context)),
+          and(isText, hasTransparentParent(device, context))
```

Comment:
Not completely sure about this. There's a theoretical change in behavior because before, if `isText(node)`  was false, it would still recurse and call `isTransparent` on the parent. That won't happen anymore. Not sure if that's okay if `node` is always guarantied to be `Element` or `Text` 🤔

Reply:
Yes, we need to keep the recursion 🤔 
```html
<div style="opacity: 0">
  <span>Hello world</span>
</div>
```
The `<span>` is transparent, because of the (non-)opacity of the `<div>`. But it seems to fail the check.

Seemsl ike we can:
* rename `hasTransparentParent` => `hasTransparentTextColor`, removing the `isTransparent` call (and typing the input as `Text`). (actually, iirc, `hasComputedStyle` does goes to parent automatically when called on text node, so we should be able to simplify that a bit)
* add `hasTransparentParent` that just calls `parent.some(isTransparent)` (might end up inlinable with a simple arrow function, which I tend to like to keep the recursion inside a single function rather than jumping between two).
* call `hasTransparentParent` as the last item in the `or` here (only recurse in last recourse).